treeherder/webapp/api/performance_data.py

```diff
@@ -473,6 +473,55 @@ class PerformanceAlertSummaryViewSet(viewsets.ModelViewSet):
     ordering = ('-created', '-id')
     pagination_class = AlertSummaryPagination
 
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.queryset)
+        pk = request.query_params.get('id')
+        page = self.paginate_queryset(queryset)
+        if page is not None:
```

Comment:
In my opinion we can exclude adding the logic for profiles when we want to use pagination, because it automatically means we don't request data for a specific summary (we don't provide an id).